x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -123,3 +123,56 @@ impl Argument for Program {
         true
     }
 }
+
+/// Represents a pointer to a block of memory allocated by the global allocator.
+/// A `Pointer` can be used directly as a param type on a `public` functions.
+///
+/// # Example
+/// ```
+/// use wasmlanche_sdk::types::Pointer;
+/// #[public]
+/// pub fn init( program: Program, name_ptr: Pointer, name_length: i64) -> bool {
```

Comment:
Would this allow us to remove `name_length` then? Do you think `Bytes` or `Memory` would be a better name here?

Reply:
Cool